acceptance/scripts/generic/testrun.sh

```diff
@@ -14,62 +14,86 @@ export BEAKER_HELPER="${BEAKER_HELPER:-acceptance/lib/helper.rb}"
 bundle install --path vendor/bundle
 
 #TODO: Someday, when our Rakefile is refactored, this section will go away
-BEAKER="bundle exec beaker init --debug"
-BEAKER="$BEAKER --type aio"
-BEAKER="$BEAKER --keyfile $BEAKER_KEYFILE"
-BEAKER="$BEAKER --helper $BEAKER_HELPER"
-BEAKER="$BEAKER --options-file $BEAKER_OPTIONS"
-BEAKER="$BEAKER --post-suite $BEAKER_POSTSUITE"
-BEAKER="$BEAKER --load-path acceptance/lib"
-
-if [ -z "$PACKAGE_BUILD_VERSION" ];
-  #TODO: curl builds.puppetlabs.lan/puppetserver.  Parse HTML, find most recent folder name.
-  then echo "$0: PACKAGE_BUILD_VERSION not set.  Please export PACKAGE_BUILD_VERSION."
-  exit -1;
-fi
+BEAKER_INIT="bundle exec beaker init --debug"
+BEAKER_INIT="$BEAKER_INIT --type aio"
+BEAKER_INIT="$BEAKER_INIT --keyfile $BEAKER_KEYFILE"
+BEAKER_INIT="$BEAKER_INIT --helper $BEAKER_HELPER"
+BEAKER_INIT="$BEAKER_INIT --options-file $BEAKER_OPTIONS"
+BEAKER_INIT="$BEAKER_INIT --post-suite $BEAKER_POSTSUITE"
+BEAKER_INIT="$BEAKER_INIT --load-path acceptance/lib"
 
 case $1 in
   -p | --p* )
+  if [ -z "$PACKAGE_BUILD_VERSION" ];
+    #TODO: Read last passing SHA file on builds.pl.d.net
+    then echo "$0: PACKAGE_BUILD_VERSION not set.  Please export PACKAGE_BUILD_VERSION."
+    exit -1;
+  fi
+
   bundle exec beaker-hostgenerator $GENCONFIG_LAYOUT > $BEAKER_CONFIG
 
-  BEAKER="$BEAKER --pre-suite $BEAKER_PRESUITE"
-  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
-  BEAKER="$BEAKER --config $BEAKER_CONFIG"
-  BEAKER="$BEAKER --preserve-hosts always"
-  $BEAKER
-  echo "Beaker exited with $?"
-  cp log/latest/hosts_preserved.yml .
+  BEAKER_INIT="$BEAKER_INIT --pre-suite $BEAKER_PRESUITE"
+  BEAKER_INIT="$BEAKER_INIT --tests $BEAKER_TESTSUITE"
+  BEAKER_INIT="$BEAKER_INIT --hosts $BEAKER_CONFIG"
+  BEAKER_INIT="$BEAKER_INIT --preserve-hosts always"
```

Comment:
Pretty sure you can get rid of this `--preserve-hosts` setting.

Reply:
Why?

Reply:
Yes probably. None of the subcommnds except `destroy` clean up the machines, regardless of failure state.

Reply:
ah. so then could the other `--preserve-hosts` below be removed as well?